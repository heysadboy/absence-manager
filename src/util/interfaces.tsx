export interface IAbsence {
    admitterId: number | null;
    admitterNote: string | null,
    confirmedAt: string | null,
    createdAt: string | null,
    crewId: number,
    endDate: string | null,
    id: number,
    memberNote: string | null,
    rejectedAt: string | null,
    startDate: string | null,
    type: string | null,
    userId: number
}

export interface IMember {
    crewId: number,
    id: number,
    image: string,
    name: string,
    userId: number
}

export interface IMemberList {
    [id: number]: IMember
}