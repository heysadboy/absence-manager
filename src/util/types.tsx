export type TPageIndex = {
    start: number;
    end: number;
    total: number
}

export type TDateRange = {
    start: string,
    end: string
}