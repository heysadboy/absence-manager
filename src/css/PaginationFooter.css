#footer {
    padding: 20px;
}

#footer-text {
    margin-top: 7px;
    font-size: medium;
}

@media screen and (max-width: 480px) {
    #footer {
        padding: 5px;
    }

    #footer-text {
        margin-top: 10px;
        font-size: small;
    }
}