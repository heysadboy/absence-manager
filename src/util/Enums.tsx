export enum EAbsenceStatus {
    requested = 'Requested',
    confirmed = 'Confirmed',
    rejected = 'Rejected'
}

export enum EPageStatus {
    loading = 'Loading',
    error = 'Error',
    success = 'Success',
    empty = 'Empty'
}

export enum EAbsenceType {
    sickness = 'Sickness',
    vacation = 'Vacation',
    all = 'All'
}
