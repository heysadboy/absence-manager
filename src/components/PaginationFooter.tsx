import { FC } from 'react';
import { TPageIndex } from '../util/Types';
import '../css/PaginationFooter.css';

interface IPaginationFooter {
    pageIndex: TPageIndex;
    setPageIndex: (pageIndex: TPageIndex) => void;
}

const PaginationFooter: FC<IPaginationFooter> = ({ pageIndex, setPageIndex }) => {

    const previousButton = () => {
        if (pageIndex.start > 0) {
            setPageIndex({ start: pageIndex.start - 10, end: pageIndex.start, total: pageIndex.total });
        }
    }

    const nextButton = () => {
        if (pageIndex.end < pageIndex.total) {
            if (pageIndex.end + 10 > pageIndex.total) {
                setPageIndex({ start: pageIndex.end, end: pageIndex.total, total: pageIndex.total });
            }
            else {
                setPageIndex({ start: pageIndex.end, end: pageIndex.end + 10, total: pageIndex.total });
            }
        }
    }

    return (
        <div id="footer" className="ui bottom fixed menu">
            <div className="ui container">
                <div id="footer-text">
                    <p>Showing <b>{pageIndex.start + 1}</b> - <b>{pageIndex.end}</b> of <b>{pageIndex.total}</b> absentees</p>
                </div>
                <div className="right menu">
                    <div className="ui basic buttons">
                        <button className="ui button" onClick={previousButton}><i className="left chevron icon"></i></button>
                        <button className="ui button" onClick={nextButton}><i className="right chevron icon"></i></button>
                    </div>
                </div>
            </div>
        </div>
    )
};

export default PaginationFooter;