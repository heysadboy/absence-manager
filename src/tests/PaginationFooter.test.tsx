import renderer from 'react-test-renderer';
import PaginationFooter from '../components/PaginationFooter';
import { TPageIndex } from '../util/Types';

test('PaginationFooter.tsx snapshot testing', () => {
    const setPageIndex = () => { }
    const pageIndex: TPageIndex = { start: 0, end: 0, total: 10 };

    const component = renderer.create(
        <PaginationFooter pageIndex={pageIndex} setPageIndex={setPageIndex} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});