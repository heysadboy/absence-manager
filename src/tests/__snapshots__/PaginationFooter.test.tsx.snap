// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationFooter.tsx snapshot testing 1`] = `
<div
  className="ui bottom fixed menu"
  id="footer"
>
  <div
    className="ui container"
  >
    <div
      id="footer-text"
    >
      <p>
        Showing 
        <b>
          1
        </b>
         - 
        <b>
          0
        </b>
         of 
        <b>
          10
        </b>
         absentees
      </p>
    </div>
    <div
      className="right menu"
    >
      <div
        className="ui basic buttons"
      >
        <button
          className="ui button"
          onClick={[Function]}
        >
          <i
            className="left chevron icon"
          />
        </button>
        <button
          className="ui button"
          onClick={[Function]}
        >
          <i
            className="right chevron icon"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
